### PR TITLE
Add TypeScript definition

### PR DESCRIPTION
**Pull request by tobenna**

### CODE REVIEW

**Comment by sindresorhus** [comment]
Thanks for creating a TypeScript definition. 🙌 Can you follow [this styleguide](https://github.com/sindresorhus/typescript-definition-style-guide)?

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+}
+
+export default class CacheableLookup {
+	constructor(options?: Options);
+	/**
+	 * DNS servers used to make the query. Can be overridden - then the new servers will be used.
+	 */
+	servers: string[];
+	/**
+	 * https://nodejs.org/api/dns.html#dns_dns_lookup_hostname_options_callback
```
Comment (szmarczak) [inline index.d.ts]:
```suggestion
	 * @see https://nodejs.org/api/dns.html#dns_dns_lookup_hostname_options_callback
```

File: index.test-d.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  expectType<ReadonlyArray<EntryObject>>(results);
+});
+
+cacheable.lookup('localhost', {all: true, details: true}, (err, results) => {
+  expectType<NodeJS.ErrnoException>(err);
+  expectType<ReadonlyArray<EntryObject & {expires: number, ttl: number}>>(results);
+});
+
+expectType<ReadonlyArray<EntryObject>>(await cacheable.query('localhost', 4));
+expectType<ReadonlyArray<EntryObject>>(await cacheable.queryAndCache('localhost', 4));
```
Comment (szmarczak) [inline index.test-d.ts]:
```suggestion
expectType<ReadonlyArray<EntryObject>>(await cacheable.queryAndCache('localhost', 4));

```

File: index.d.ts [hunk 3]
```diff
@@ -0,0 +1,100 @@
+import Keyv from "keyv";
```
Comment (szmarczak) [inline index.d.ts]:
```suggestion
import Keyv from 'keyv';
```

File: index.d.ts [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+import {Resolver, LookupAddress} from "dns";
```
Comment (szmarczak) [inline index.d.ts]:
```suggestion
import {Resolver, LookupAddress} from 'dns';
```

File: index.d.ts [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+export default class CacheableLookup {
+	constructor(options?: Options);
+	/**
+	 * DNS servers used to make the query. Can be overridden - then the new servers will be used.
+	 */
+	servers: string[];
+	/**
+	 * https://nodejs.org/api/dns.html#dns_dns_lookup_hostname_options_callback
+	 */
+	lookup(hostname: string, family: IPFamily, callback: (err: NodeJS.ErrnoException, address: string, family: IPFamily) => void): void;
```
Comment (szmarczak) [inline index.d.ts]:
```suggestion
	lookup(hostname: string, family: IPFamily, callback: (error: NodeJS.ErrnoException, address: string, family: IPFamily) => void): void;
```

same else

**Comment by tobenna** [comment]
Hi @szmarczak, is there anything else I need to look into with regard to this?

**Comment by szmarczak** [comment]
Amazing work! Thank you :) I'll do a new release today afternoon.

**Pull request merged**